client/checkpoint_delete.go

```diff
@@ -2,11 +2,13 @@ package client
 
 import (
 	"golang.org/x/net/context"
+
+	"github.com/docker/engine-api/types"
 )
 
 // CheckpointDelete deletes the checkpoint with the given name from the given container
-func (cli *Client) CheckpointDelete(ctx context.Context, containerID string, checkpointID string) error {
-	resp, err := cli.delete(ctx, "/containers/"+containerID+"/checkpoints/"+checkpointID, nil, nil)
+func (cli *Client) CheckpointDelete(ctx context.Context, containerID string, options types.CheckpointDeleteOptions) error {
+	resp, err := cli.delete(ctx, "/containers/"+containerID+"/checkpoints/"+options.CheckpointID, nil, nil)
```

Comment:
@WeiZhang555 I think `checkpointID` on `CheckpointDelete` is required, so it should not be changed in an options (just `ContainerStart` imho).